docs/lab/01.md

```diff
@@ -3,3 +3,467 @@ sidebar_position: 1
+        sign_in_count: 1,
+    };
+ 
+    user1.email = String::from("anotheremail@example.com")
+}
+```
+
+> Note that the entire instance must be **editable**  ; Rust **doesn't allow us** to mark only certain fields as mutable!
+
+As with any expression, we can construct a new instance of the structure as the last expression in the function body to implicitly return this new instance.
+
+```rust
+fn build_user(email: String, username: String) -> User {
+    User {
+        active: true,
+        username: username,
+        email: email,
+        sign_in_count: 1,
+    }
+}
+```
+- ### Tuple structs
+
+Rust also supports structures that resemble tuples, called **tuple structs** . Tuple structures have the additional meaning provided by the structure name but do not have names associated with their fields; instead, they just have the field types. Tuple structures are useful when you want to name the entire tuple and make it a different type from other tuples, and when naming each field as in a regular structure would be wordy or redundant.
+
+```rust
+struct Color(i32, i32, i32);
+struct Point(i32, i32, i32);
+ 
+fn main() {
+    let black = Color(0, 0, 0);
+    let origin = Point(0, 0, 0);
+}
+```
+
+
+> To understand better, read [chapter 5](https://doc.rust-lang.org/book/ch05-00-structs.html) of the documentation.
+
+### String
+Rust has only one type of string in the base language, which is the string slice str which is usually seen in its borrowed form **&str** .
+
+The **String** type , which is provided by the Rust standard library rather than encoded in the main language, is a scalable, mutable, and owned UTF-8 encoded string type .
+
+- ### Creating a new String
+```rust
+    let mut s = String::new();
+```
+This line creates a new empty string called `s`, which we can then load data into.
+
+**We can use the String::from** function or the **to_string()** function to create a string from a string literal:
+```rust
+    let s = String::from("initial contents");
+```
+```rust
+let data = "initial contents";
+ 
+    let s = data.to_string();
+ 
+    // the method also works on a literal directly:
+    let s = "initial contents".to_string();
+```
+
+- ### Adding to a stream
+
+We can expand a string using the **push_str** method to add a string slice.
+```rust
+let mut s = String::from("foo");
+s.push_str("bar");
+```
+**The push** method takes **a single character** as a parameter and adds it to the string.
+
+```rust
+    let mut s = String::from("lo");
+    s.push('l');
+```
+- ### Iteration Methods on Strings
+
+The best way to operate on pieces of strings is to be explicit about whether you want characters or bytes. For individual Unicode scalar values, use the **chars** method .
+```rust
+for c in "Зд".chars() {
+    println!("{c}");
+}
+```
+### Run the program
+We must access the director where the main.rs file is located and execute the command:
+
+```
+cargo run
+```
+
+## Exercises
```

Comment:
Should we add points to these?

Reply:
2p given by default
1. - 1p
2. - 1p
3. - 1p
4. - 1p
5. - 2p
6. - 2p

Reply:
Out of 10?

Reply:
I would grade every lab out of 10